index.js

```diff
@@ -1,84 +1,170 @@
+require('dotenv').config();
 var fs = require('fs');
-var http = require('http');
-var https = require('https');
 var request = require('request');
-var emoji = require('node-emoji')
 
-const {
-    IncomingWebhook
-} = require('@slack/client');
-const url = process.env.SLACK_WEBHOOK_URL;
+const UrlStatus = {
+    Up: 'Up',
+    Pending: 'Pending',
+    StillDown: 'StillDown',
+};
 
-const webhookurl = "https://hooks.slack.com/services/<some-sort-of-secret-key>";
-const webhook = new IncomingWebhook(webhookurl);
-var notReachable = {};
-var notReachabletwice = {};
+Object.freeze(UrlStatus);
 
+const webhookUrl = process.env.SLACK_WEBHOOK_URL;
+const urlContainer = {};
+const urlErrorContainer = {};
 
 /* Read configuration and start loops */
 fs.readFile('./config.json', 'utf8', function (err, data) {
     if (err) throw err;
-    var conf = JSON.parse(data);
-    var urls = conf.urls;
+    const conf = JSON.parse(data);
+    const urls = conf.urls;
 
-    for (var name in urls) {
-        var url = urls[name].url;
-        var interval = urls[name].interval;
-        createloop(name, url, interval);
+    for (const [name, params] of Object.entries(urls)) {
+        const url = params.url;
+        const interval = params.interval;
+        const dateBounds = {
+            minTimeOfDay: params.minTimeOfDay,
+            maxTimeOfDay: params.maxTimeOfDay,
+            daysOfTheWeek: params.daysOfTheWeek
+        }
+        const mentionsWhenDown = params.mentionsWhenDown;
+        createLoop(name, url, interval, dateBounds, mentionsWhenDown);
```

Comment:
it's better to pass a single `params` to `createLoop`, there you do the extraction of what's used